src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -981,19 +982,22 @@ public GHPermissionType fetch(String username) throws IOException, InterruptedEx
                                             public SCMRevision create(@NonNull PullRequestSCMHead head,
                                                                       @Nullable Void ignored)
                                                     throws IOException, InterruptedException {
+                                                String baseHash = pr.getBase().getSha();
+                                                String mergeHash = null;
+
                                                 switch (strategy) {
                                                     case MERGE:
                                                         request.checkApiRateLimit();
-                                                        GHRef mergeRef = ghRepository.getRef(
-                                                                "heads/" + pr.getBase().getRef()
-                                                        );
-                                                        return new PullRequestSCMRevision(head,
-                                                                mergeRef.getObject().getSha(),
-                                                                pr.getHead().getSha());
+                                                        baseHash = ghRepository.getRef("heads/" + head.getTarget().getName()).getObject().getSha();
+                                                        mergeHash = pr.getMergeCommitSha();
```

Comment:
So this is going to produce a synthetic commit which may or may not have as parents the `baseHash` and the `.head.sha`. Depends on exactly when you call it.

Reply:
Again, I'm not sure what you're pointing out here or why.